### PR TITLE
Adding of clear and cancel button in create preset form

### LINKED ISSUE #8902: Add a clear and cancel button in the preset pop-up of asset camera configuration page
**Describe the bug**

When trying to clear or close the pop-up on the camera asset configuration page, there is no clear or cancel option available, causing confusion for users.

**To Reproduce**

1. Go to asset tab
2. Click on any camera asset card
3. click on configuration and try to create a preset

**Expected behavior**

- [ ] Add a Cancel button adjacent to the Submit button for closing the pop-up.
- [ ] Add a Clear button within the preset input field to clear the input value.

Note: Ensure these changes align with the existing design language for consistency.

**Screenshots**

Image

### PR DESCRIPTION
**Pull request by Rishith25**
## Proposed Changes

- Fixes #8902 
- Added a Cancel button adjacent to the Submit button for closing the pop-up.
- Added a Clear button within the preset input field to clear the input value.

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

Screenshot 2024-10-24 223310

### CODE REVIEW

**Review by shivankacker — APPROVED** [review]

**Review by bodhish — APPROVED** [review]
LGTM

File: src/components/CameraFeed/ConfigureCamera.tsx [hunk 1]
```diff
@@ -389,16 +389,37 @@ export default function ConfigureCamera(props: Props) {
...
+                        onClick={() => setPresetName("")}
+                      >
+                        <CareIcon icon="l-times-circle" className="text-lg" />
+                      </button>
+                    )}
+                  </div>
+
+                  <div className="cui-form-button-group mt-4 flex w-full flex-col justify-end space-y-2 sm:flex-row sm:space-x-2 sm:space-y-0">
+                    <Cancel
+                      shadow={false}
```
Comment (rithviknishad) [inline src/components/CameraFeed/ConfigureCamera.tsx]:
```suggestion
```

**Review by rithviknishad — APPROVED** [review]
LGTM; just curious why shadow was disabled for the cancel button

File: src/components/CameraFeed/ConfigureCamera.tsx (hunk 1 above)
Comment (Rishith25) [reply src/components/CameraFeed/ConfigureCamera.tsx]: Done the changes. I thought to maintain a simpler and cleaner design.

**Comment by Rishith25** [comment]
@rithviknishad Can you please assign 2 story points for this

**Comment by bodhish** [comment]
@Rishith25 for minor changes we keep it to 1.

**Comment by nihal467** [comment]
image

- [ ] Reposition the clear button properly in the pop-up, rest looks good

**Comment by Rishith25** [comment]
@nihal467 Done the changes of clear button alignment

File: src/components/CameraFeed/ConfigureCamera.tsx [hunk 2]
```diff
@@ -389,16 +389,28 @@ export default function ConfigureCamera(props: Props) {
...
+                      value={presetName}
+                      onChange={({ value }) => setPresetName(value)}
+                      errorClassName="hidden"
+                      placeholder={t("preset_name_placeholder")}
+                      suggestions={presetNameSuggestions}
+                      isClear={true}
+                    />
+                  </div>
+
+                  <div className="cui-form-button-group mt-4 flex w-full flex-col justify-end space-y-2 sm:flex-row sm:space-x-2 sm:space-y-0">
```
Comment (rithviknishad) [inline src/components/CameraFeed/ConfigureCamera.tsx]:
```suggestion
                  <div className="cui-form-button-group">
```

File: src/components/CameraFeed/ConfigureCamera.tsx [hunk 3]
```diff
@@ -389,16 +389,28 @@ export default function ConfigureCamera(props: Props) {
...
+                    <Cancel
+                      onClick={() => {
+                        setCreatePreset(undefined);
+                        setPresetName("");
+                      }}
+                      className="ml-0 w-full sm:ml-2 sm:w-auto"
```
Comment (rithviknishad) [inline src/components/CameraFeed/ConfigureCamera.tsx]:
```suggestion
```

File: src/components/CameraFeed/ConfigureCamera.tsx [hunk 4]
```diff
@@ -417,6 +429,7 @@ export default function ConfigureCamera(props: Props) {
                         cameraPresetsQuery.refetch();
                       }}
                       disabled={!presetName}
+                      className="w-full sm:w-auto"
```
Comment (rithviknishad) [inline src/components/CameraFeed/ConfigureCamera.tsx]:
```suggestion
```

File: src/components/Form/FormFields/TextFormField.tsx [hunk 5]
```diff
@@ -35,30 +36,47 @@ const TextFormField = forwardRef((props: TextFormFieldProps, ref) => {
   const hasTrailing = !!(trailing || trailingFocused);
   const hasIcon = hasLeading || hasTrailing;
   const [showPassword, setShowPassword] = useState(false);
+  const isClear = props.isClear || false;
```
Comment (rithviknishad) [inline src/components/Form/FormFields/TextFormField.tsx]: null/undefined is falsey, it's no-op when used in conditional statements.

```suggestion
```

File: src/components/CameraFeed/ConfigureCamera.tsx [hunk 6]
```diff
@@ -389,16 +389,28 @@ export default function ConfigureCamera(props: Props) {
...
+                      name="preset-name"
+                      className="w-full py-4"
+                      value={presetName}
+                      onChange={({ value }) => setPresetName(value)}
+                      errorClassName="hidden"
+                      placeholder={t("preset_name_placeholder")}
+                      suggestions={presetNameSuggestions}
+                      isClear={true}
+                    />
+                  </div>
```
Comment (rithviknishad) [inline src/components/CameraFeed/ConfigureCamera.tsx]: why wrap in a div with `relative` class here? is this needed?

File: src/components/Form/FormFields/TextFormField.tsx [hunk 7]
```diff
@@ -24,6 +24,7 @@ export type TextFormFieldProps = FormFieldBaseProps<string> &
     trailingPadding?: string | undefined;
     leadingPadding?: string | undefined;
     suggestions?: string[];
+    isClear?: boolean;
```
Comment (rithviknishad) [inline src/components/Form/FormFields/TextFormField.tsx]: `isClear` doesn't make sense. One may not understand what this prop does by simply reading it.

```suggestion
    clearable?: boolean;
```

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Comment by Rishith25** [comment]
@rithviknishad Changes are done

File: src/components/CameraFeed/ConfigureCamera.tsx [hunk 8]
```diff
@@ -391,14 +391,23 @@ export default function ConfigureCamera(props: Props) {
...
                   />
+
                   <div className="cui-form-button-group">
+                    <Cancel
+                      onClick={() => {
+                        setCreatePreset(undefined);
+                        setPresetName("");
+                      }}
+                    />
+
```
Comment (rithviknishad) [inline src/components/CameraFeed/ConfigureCamera.tsx]:
```suggestion
```

File: src/components/Form/FormFields/TextFormField.tsx [hunk 9]
```diff
@@ -40,25 +41,41 @@ const TextFormField = forwardRef((props: TextFormFieldProps, ref) => {
     return showPassword ? "text" : "password";
   };
 
+  const handleClear = () => {
+    field.handleChange("");
+  };
+
```
Comment (rithviknishad) [inline src/components/Form/FormFields/TextFormField.tsx]:
```suggestion
```

File: src/components/Form/FormFields/TextFormField.tsx [hunk 10]
```diff
@@ -40,25 +41,41 @@ const TextFormField = forwardRef((props: TextFormFieldProps, ref) => {
...
+        name={field.name}
+        value={field.value}
+        required={field.required}
+        onChange={(e) => field.handleChange(e.target.value)}
+      />
+      {props.clearable && field.value && (
+        <button
+          type="button"
+          className="absolute right-2 top-1/2 -translate-y-1/2 transform text-gray-500"
+          onClick={handleClear}
```
Comment (rithviknishad) [inline src/components/Form/FormFields/TextFormField.tsx]:
```suggestion
          onClick={() => field.handleChange("")}
```

**Review by rithviknishad — CHANGES_REQUESTED** [review]
LGTM; just some cleanups

**Comment by Rishith25** [comment]
@rithviknishad The cleanups are done

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
LGTM

**Pull request merged**